packages/db-base/src/lib/inMemFileDB.ts

```diff
@@ -39,12 +32,83 @@ const tools = require('./tools.js');
 // };
 //
 
+interface ConnectionOptions {
+    // relative path to the data dir
+    backup?: BackupOptions;
+    dataDir: string;
+}
+
+type ChangeFunction = (id: string, state: any) => void;
+
+interface DbStatus {
+    type: string;
+    server: boolean;
+}
+
+interface BackupOptions {
+    // deactivates backup if false
+    disabled: boolean;
+    // minimum number of files
+    files: number;
+    hours: number;
+    // minutes
+    period: number;
+    path: string;
+}
+
+interface DbOptions {
+    backupDirName: string;
+    fileName: string;
+}
+
+interface FileDbSettings {
+    fileDB: DbOptions;
+    jsonlDB: DbOptions;
+    backup: BackupOptions;
+    change?: ChangeFunction;
+    connected: (nameOfServer: string) => void;
+    logger: tools.Logger;
+    connection: ConnectionOptions;
+    // unused
+    auth: null;
+    secure: boolean;
+    // as required by createServer
+    certificates: any;
+    port: number;
+    host: string;
+    // logging namespace
+    namespace?: string;
+}
+
+interface Subscription {
+    pattern: string;
+    regex: RegExp;
+    options: any;
+}
+
+interface SubscriptionClient {
+    _subscribe?: Record<string, Subscription[]>;
+}
+
 /**
  * The parent of the class structure, which provides basic JSON storage
  * and general subscription and publish functionality
  **/
-class InMemoryFileDB {
-    constructor(settings) {
+export class InMemoryFileDB {
+    private settings: FileDbSettings;
+    private readonly change: ChangeFunction | undefined;
+    private dataset: Record<string, any>;
+    private readonly namespace: string;
+    private lastSave: null | number;
+    private zlib: null | typeof import('zlib');
```

Comment:
Do we really need the dynamic import of `zlib`? Doesn't seem like anything works properly if we cannot import it. Just importing it at the top would make the code a bit easier.